### PR TITLE
Rerender for newer GCC & Mamba

### LINKED ISSUE #80: libgfortran issue downstream
Since the `mamba` upgrade in October, I see the following concretization issue in my downstream project:
```
The reported errors are:
- Encountered problems while solving:
-   - package fftw-3.3.10-nompi_h427eb53_101 requires libgfortran5 >=9.4.0, but none of the providers can be installed
```

My project is a plain C/C++ project:
https://github.com/conda-forge/warpx-feedstock/pull/33
and and declares the FFTW dependency here:
https://github.com/conda-forge/warpx-feedstock/blob/fe5437ec5282ebdbf4e43af96b8a894e26debb31/recipe/meta.yaml#L43

Should we try to rerender the feedstock and see if this solves it?

### PR DESCRIPTION
**Pull request by ax3l**
Trying to close #80 by rerendering the feedstock with latest compilers and mamba (default since [Oct](https://conda-forge.org/docs/user/announcements.html#announcements), fftw not rerendered since Sep).

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
@jakirkham let's see if this fixes it :) Ready for review & `automerge` label :)

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Axel! 😄

**Comment by duncanmmacleod** [comment]
It would be good to include my changes from #82 here (if only to avoid another build).

**Comment by egpbos** [comment]
Weird, I tried to rerender your PR @duncanmmacleod, since it says it was rerendered with conda-build 3.21.4, whereas this PR was rerendered with 3.21.6.

Does anyone understand why a newer rerender (the one in #82 was done 2 hours ago, the one in this PR was done 18 hours ago...) would use an older conda-build version?

**Comment by egpbos** [comment]
@conda-forge-admin, please rerender

**Comment by duncanmmacleod** [comment]
> Weird, I tried to rerender your PR @duncanmmacleod, since it says it was rerendered with conda-build 3.21.4, whereas this PR was rerendered with 3.21.6.
> 
> Does anyone understand why a newer rerender (the one in #82 was done 2 hours ago, the one in this PR was done 18 hours ago...) would use an older conda-build version?

I ran the rerender on my own machine, I must just have an older version of conda-build.

**Comment by egpbos** [comment]
@ax3l could you allow @duncanmmacleod to cherry-pick and push his Python update commit to this branch?

**Comment by duncanmmacleod** [comment]
> @ax3l could you allow @duncanmmacleod to cherry-pick and push his Python update commit to this branch?

alternatively, @ax3l could run

```bash
curl -Ls https://github.com/conda-forge/fftw-feedstock/pull/82/commits/3f461dafad8b3f3298a573c4226a1139eeb0f177.patch | git am -S
```

to pull my commit onto this branch; then regenerate the feedstock.

**Comment by ax3l** [comment]
Patch from #82 applied, thanks!

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Review by egpbos — APPROVED** [review]
Super, thanks!

**Comment by ax3l** [comment]
Is it possible the `aarch64` builds are too slow currently, due to emulation? Looks like they ran 6 hrs on Azure.

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Review by beckermr — CHANGES_REQUESTED** [review]
We should adjust the build script to run tests under qemu for the cross-compiled stuff on linux

**Comment by ax3l** [comment]
Looks like tests are only available at build time already in this repo.

**Comment by beckermr** [comment]
> Looks like tests are only available at build time already in this repo / in fftws logic, aren't they?

Yes. There are lines in the build.sh that turn them off for some platforms. We simply should remove those lines.

**Comment by beckermr** [comment]
This bit too?

```
 # Disable Tests since we don't have enough time on Azure
  if [[ "$CI" == "azure" ]]; then
    TEST_CMD=""
  fi
```

**Comment by beckermr** [comment]
The tests did not run. Trying this:

@conda-forge-admin rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -67,7 +67,7 @@ if [[ "$target_platform" == "osx-arm64" ]]; then
   DISABLE_LONG_DOUBLE=1
 fi
 
-if [[ "$CONDA_BUILD_CROSS_COMPILATION" == 1 ]]; then
+if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" == 1 && "${CROSSCOMPILING_EMULATOR:-}" == "" ]]; then
```
Comment (beckermr) [inline recipe/build.sh:70]:
```suggestion
if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" == 1 && "${CROSSCOMPILING_EMULATOR:-}" != "" ]]; then
```

**Comment by beckermr** [comment]
ahhhhh!

This build is installing an ancient ci-setup version?

```
conda-forge/linux-64::conda-forge-ci-setup-3.13.0-py38h7bc8238_0
```

**Comment by beckermr** [comment]
I think it might be the boa run constraint? The boa version is 0.6.2.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: conda-forge.yml [hunk 2]
```diff
@@ -1,6 +1,12 @@
-azure: {timeout_minutes: 600}
-build_platform: {osx_arm64: osx_64}
+azure:
+  timeout_minutes: 600
+build_platform:
+  linux_aarch64: linux_64
+  osx_arm64: osx_64
 conda_forge_output_validation: true
 idle_timeout_minutes: 90
```
Comment (beckermr) [inline conda-forge.yml:7]:
```suggestion
# idle_timeout_minutes: 90
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/build.sh [hunk 3]
```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+echo "\n\n============================================"
+env
+echo "============================================\n\n"
+
+
```
Comment (beckermr) [inline recipe/build.sh:7]:
```suggestion
```

File: recipe/build.sh [hunk 4]
```diff
@@ -83,6 +92,10 @@ if [[ "$DISABLE_LONG_DOUBLE" != 1 ]]; then
     build_cases+=("$CONFIGURE --enable-long-double ${ARCH_OPTS_LONG_DOUBLE}")
 fi
 
+echo "\n\n============================================"
+echo "test command: ${TEST_CMD}"
+echo "============================================\n\n"
```
Comment (beckermr) [inline recipe/build.sh:97]:
```suggestion
echo " "
echo "============================================"
echo "============================================"
echo "test command: ${TEST_CMD}"
echo "============================================"
echo "============================================"
echo " "
```

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
huzzah!

**Comment by ax3l** [comment]
Thank you for your help! :)

**Pull request merged**